### DISCUSSION TITLE
Show real production costs for stock items

### DISCUSSION THREAD

**Discussion by lippoliv** [inventree/InvenTree#5547] [General]
First of all: Nice tool. I'm just getting into it, evaluate if it fits our needs, but it looks good. Thanks to all contributors :)

Currently in the stock overview, there's the column `Stock Value` which shows the real purchase price for purchased items, but an min - max price for items which are build within the company. I wonder if we can make the meaning of the column "equal".

We can calculate the production cost price per stock item, since we know the BOM and allocate other stock to it. We maybe even know "internal prices", which probably come into play here as well.
So per stock, we can tell the purchase price, and as long as BOM is configured (where we get min - max price from anyhow) we also can tell per stock what it did cost to produce (at least material).

Benefits
1. If we add this field, we can show it in the overview, so everybody would be able to calculate the raw stock total value.
2. For orders, we can calculate exactly, what is our material costs in this order. 
3. For parts we can add a historical production prices chart, so we visually see changes and can adapt our selling prices to it.

I searched if it was discussed already, haven't found something.
I searched if this is implemented as a plugin, but haven't found any.

What are your thoughts on that?

**Comment by matmair** [comment]
It would be possible but pricing is already pretty complex so I am not sure how fast that can be implemented.

It is hard to determine an absolute value for a piece as the price for source material normally fluctuates and pin-pointing a few specific stock pieces creates false certainty of a final price which probably is just false.

There are already charts for various price indicators over time on the part given the caveats mentioned above I am not sure what value an additional graph brings.

**Reply by lippoliv** [reply]
I'm new to InvenTree, so may I got this wrong, buy I would say the way the models are layed out we should be precisely able to tell the price:

1. We have `Part`s
2. We purchase of build them
3. If it's a `Part` that we can build, we describe BOM (at this moment we can just give a price indication)
4. Whenever we purchase `Parts`, `Stock` is created (now we have a price). There's multiple `Stock`s for a `Part`, that's why the price does vary
5. Now we create a `Build` for a `Part`
6. Next we allocate one or more `Stock` to the `Build` (now we exactly now the price of the `Build`)
7. And as we know how many `Stock` we produce in this `Build`, we can calculate the price for the `Stock`

What do I miss?

**Reply by SchrodingersGat** [reply]
So specifically the additional functionality you are looking for is to calculate the "cost to build" of a build output. 

This makes sense and would be a good feature with the following caveats:

- getting accurate pricing data is already tricky as Matthias outlined above
- where explicit stock price is not available we use a range of the available data
- if the BOM is incomplete, obviously calculated price would be wrong
- what about manufacturing costs (labour etc) associated with the build order?

I am ok with the idea but it would need to be very carefully considered

**Reply by lippoliv** [reply]
Thanks for the quick feedback.

I agree: "Shit in, shit out", right? If I misspell quantity in a hurry, I will get wron min-max range as well, right? :)



I guess I still do not fully understand this
> getting accurate pricing data is already tricky as Matthias outlined above

Can you elaborate a little more?
